src/EquivalenceCheckingManager.cpp

```diff
@@ -479,6 +524,15 @@ namespace ec {
                 break;
             }
 
+            if (dynamic_cast<ZXEquivalenceChecker*>(checker)) {
+                results.equivalence = result;
+                if (result == EquivalenceCriterion::EquivalentUpToGlobalPhase || (result == EquivalenceCriterion::ProbablyNotEquivalent && !(runAlternating || runConstruction || runSimulation))) {
```

Comment:
Maybe it would make sense to extract this into a method `onlyZXCheckerConfigured`? It's used a couple of times.